### PR TITLE
Merge unit tests part4

### PR DESCRIPTION
**Pull request by OHerasym**
## PR provides following changes:

Fix protocol_handler tests
Fix transport manager tests
Fix security manager unit tests
Correct tests according to code changes and make tests buildable

> Tests builds on LINUX & WINDOWS & QT platform

Related [task](https://adc.luxoft.com/jira/browse/APPLINK-23755)

Disabled tests:
- protocol_packet_test.cc:214:TEST_F(ProtocolPacketTest, DISABLED_DeserializeZeroPacket) {\* 
- protocol_handler_tm_test.cc:302:TEST_F(ProtocolHandlerImplTest, DISABLED_RecieveEmptyRawMessage) {
- protocol_handler_tm_test.cc:308:TEST_F(ProtocolHandlerImplTest, DISABLED_RecieveOnUnknownConnection) {
- protocol_handler_tm_test.cc:327:       DISABLED_StartSession_Unprotected_SessionObserverReject) {
- protocol_handler_tm_test.cc:370:       DISABLED_StartSession_Protected_SessionObserverReject) {
- protocol_handler_tm_test.cc:417:       DISABLED_StartSession_Unprotected_SessionObserverAccept) {
- protocol_handler_tm_test.cc:448:       DISABLED_StartSession_Protected_SessionObserverAccept) {
- protocol_handler_tm_test.cc:459:TEST_F(ProtocolHandlerImplTest, DISABLED_EndSession_SessionObserverReject) {
- protocol_handler_tm_test.cc:484:TEST_F(ProtocolHandlerImplTest, DISABLED_EndSession_Success) {
- protocol_handler_tm_test.cc:513:       DISABLED_SecurityEnable_StartSessionProtocoloV1) {
- protocol_handler_tm_test.cc:550:       DISABLED_SecurityEnable_StartSessionUnprotected) {
- protocol_handler_tm_test.cc:579:       DISABLED_SecurityEnable_StartSessionProtected_Fail) {
- protocol_handler_tm_test.cc:614:       DISABLED_SecurityEnable_StartSessionProtected_SSLInitialized) {
- protocol_handler_tm_test.cc:658:       DISABLED_SecurityEnable_StartSessionProtected_HandshakeFail) {
- protocol_handler_tm_test.cc:725:       DISABLED_SecurityEnable_StartSessionProtected_HandshakeSuccess) {
- protocol_handler_tm_test.cc:795:    DISABLED_SecurityEnable_StartSessionProtected_HandshakeSuccess_ServiceProtectedBefore) {
- protocol_handler_tm_test.cc:865:    DISABLED_SecurityEnable_StartSessionProtected_HandshakeSuccess_SSLIsNotPending) {
- protocol_handler_tm_test.cc:934:TEST_F(ProtocolHandlerImplTest, DISABLED_FloodVerification) {
- protocol_handler_tm_test.cc:963:TEST_F(ProtocolHandlerImplTest, DISABLED_FloodVerification_ThresholdValue) {
- protocol_handler_tm_test.cc:991:TEST_F(ProtocolHandlerImplTest, DISABLED_FloodVerification_VideoFrameSkip) {
- protocol_handler_tm_test.cc:1012:TEST_F(ProtocolHandlerImplTest, DISABLED_FloodVerification_AudioFrameSkip) {
- protocol_handler_tm_test.cc:1034:TEST_F(ProtocolHandlerImplTest, DISABLED_FloodVerificationDisable) {
- protocol_handler_tm_test.cc:1057:TEST_F(ProtocolHandlerImplTest, DISABLED_MalformedVerificationDisable) {
- protocol_handler_tm_test.cc:1083:TEST_F(ProtocolHandlerImplTest, DISABLED_MalformedLimitVerification) {
- protocol_handler_tm_test.cc:1123:       DISABLED_MalformedLimitVerification_MalformedStock) {
- protocol_handler_tm_test.cc:1189:       DISABLED_MalformedLimitVerification_MalformedOnly) {
- protocol_handler_tm_test.cc:1245:       DISABLED_MalformedLimitVerification_NullTimePeriod) {
- protocol_handler_tm_test.cc:1273:TEST_F(ProtocolHandlerImplTest, DISABLED_MalformedLimitVerification_NullCount) {
- protocol_handler_tm_test.cc:1302:       DISABLED_SendEndServicePrivate_NoConnection_MessageNotSent) {
- protocol_handler_tm_test.cc:1315:       DISABLED_SendEndServicePrivate_EndSession_MessageSent) {
- protocol_handler_tm_test.cc:1335:       DISABLED_SendEndServicePrivate_ServiceTypeControl_MessageSent) {
- protocol_handler_tm_test.cc:1354:TEST_F(ProtocolHandlerImplTest, DISABLED_SendHeartBeat_NoConnection_NotSent) {
- protocol_handler_tm_test.cc:1366:TEST_F(ProtocolHandlerImplTest, DISABLED_SendHeartBeat_Successful) {
- protocol_handler_tm_test.cc:1385:TEST_F(ProtocolHandlerImplTest, DISABLED_SendHeartBeatAck_Successful) {
- protocol_handler_tm_test.cc:1407:       DISABLED_SendHeartBeatAck_WrongProtocolVersion_NotSent) {
- protocol_handler_tm_test.cc:1432:       DISABLED_SendMessageToMobileApp_SendSingleControlMessage) {
- protocol_handler_tm_test.cc:1464:       DISABLED_SendMessageToMobileApp_SendSingleNonControlMessage) {
- protocol_handler_tm_test.cc:1497:       DISABLED_SendMessageToMobileApp_SendMultiframeMessage) {
- protocol_payload_test.cc:234:     DISABLED_ExtractProtocolWithJSONWithDataWithWrongPayloadSize) {
- multiframe_builder_test.cc:425:TEST_F(MultiFrameBuilderTest, DISABLED_Add_ConsecutiveFrames_per1) {
- transport_manager_default_test.cc:45:TEST(TestTransportManagerDefault, DISABLED_Init_LastStateNotUsed) {
- transport_manager_default_test.cc:59:TEST(TestTransportManagerDefault, DISABLED_Init_LastStateUsed) {
- transport_manager_impl_test.cc:336:TEST_F(TransportManagerImplTest, DISABLED_SearchDevices_AdaptersNotAdded) {
- transport_manager_impl_test.cc:344:TEST_F(TransportManagerImplTest, DISABLED_AddTransportAdapterSecondTime) {
- transport_manager_impl_test.cc:349:TEST_F(TransportManagerImplTest, DISABLED_ConnectDevice) {
- transport_manager_impl_test.cc:471:TEST_F(TransportManagerImplTest, DISABLED_SearchDevices_DeviceNotFound) {
- transport_manager_impl_test.cc:480:TEST_F(TransportManagerImplTest, DISABLED_SearchDevices_AdapterNotSupported) {
- transport_manager_impl_test.cc:489:TEST_F(TransportManagerImplTest, DISABLED_SearchDevices_AdapterWithBadState) {
- transport_manager_impl_test.cc:498:TEST_F(TransportManagerImplTest, DISABLED_SendMessageToDevice) {
- transport_manager_impl_test.cc:664:       DISABLED_ReceiveEventFromDevice_OnSearchDeviceDone) {
- transport_manager_impl_test.cc:683:       DISABLED_ReceiveEventFromDevice_OnSearchDeviceFail) {
- transport_manager_impl_test.cc:702:       DISABLED_ReceiveEventFromDevice_DeviceListUpdated) {
- transport_manager_impl_test.cc:888:TEST_F(TransportManagerImplTest, DISABLED_HandleMessage_ConnectionNotExist) {
- transport_manager_impl_test.cc:906:       DISABLED_SetVisibilityOn_TransportAdapterNotSupported) {
- transport_manager_impl_test.cc:913:       DISABLED_SetVisibilityOff_TransportAdapterNotSupported) {
- tcp_transport_adapter_test.cc:74:TEST_F(TcpAdapterTest, DISABLED_StoreDataWithOneDeviceAndOneApplication) {
- tcp_transport_adapter_test.cc:117:TEST_F(TcpAdapterTest, DISABLED_StoreDataWithSeveralDevicesAndOneApplication) {
- tcp_transport_adapter_test.cc:173:       DISABLED_StoreDataWithSeveralDevicesAndSeveralApplications) {
- tcp_transport_adapter_test.cc:274:TEST_F(TcpAdapterTest, DISABLED_StoreDataWithOneDevice_RestoreData) {
- tcp_transport_adapter_test.cc:310:TEST_F(TcpAdapterTest, DISABLED_StoreDataWithSeveralDevices_RestoreData) {
- ssl_certificate_handshake_test.cc:355:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_ServerSide) {
- ssl_certificate_handshake_test.cc:372:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_ServerSide_NoCACertificate) {
- ssl_certificate_handshake_test.cc:399:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_ClientSide) {
- ssl_certificate_handshake_test.cc:416:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_ClientSide_NoCACertificate) {
- ssl_certificate_handshake_test.cc:445:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_BothSides) {
- ssl_certificate_handshake_test.cc:479:TEST_F(SSLHandshakeTest, DISABLED_ExpiredCert) {
- ssl_certificate_handshake_test.cc:497:TEST_F(SSLHandshakeTest, DISABLED_AppNameAndAppIDInvalid) {
- ssl_certificate_handshake_test.cc:550:TEST_F(SSLHandshakeTest, DISABLED_CAVerification_BothSides_ResetConnection) {
- security_query_test.cc:138:TEST_F(SecurityQueryTest, DISABLED_QueryConstructor) {
- security_query_test.cc:156:TEST_F(SecurityQueryTest, DISABLED_QueryConstructor2) {
- security_query_test.cc:223:TEST_F(SecurityQueryTest, DISABLED_Parse_NullData) {
- security_query_test.cc:238:TEST_F(SecurityQueryTest, DISABLED_Parse_LessHeaderData) {
- security_query_test.cc:255:TEST_F(SecurityQueryTest, DISABLED_Parse_HeaderData) {
- security_query_test.cc:276:TEST_F(SecurityQueryTest, DISABLED_Parse_HeaderDataWrong) {
- security_query_test.cc:300:TEST_F(SecurityQueryTest, DISABLED_Parse_InvalidQuery) {
- security_query_test.cc:336:TEST_F(SecurityQueryTest, DISABLED_Parse_InvalidQuery_UnknownTypeId) {
- security_query_test.cc:364:TEST_F(SecurityQueryTest, DISABLED_Parse_InvalidQuery_UnknownId_Response) {
- security_query_test.cc:389:TEST_F(SecurityQueryTest, DISABLED_Parse_Handshake) {
- security_query_test.cc:423:TEST_F(SecurityQueryTest, DISABLED_Parse_InternalError) {
- ssl_context_test.cc:365:TEST_F(SSLTest, DISABLED_OnTSL2Protocol_BrokenHandshake) {
- ssl_context_test.cc:384:TEST_F(SSLTest, DISABLED_OnTSL2Protocol_Positive) {
- ssl_context_test.cc:439:TEST_F(SSLTest, DISABLED_OnTSL2Protocol_EcncryptionFail) {
- crypto_manager_impl_test.cc:137:TEST_F(CryptoManagerTest, DISABLED_WrongInit) {
- security_manager_test.cc:175:TEST_F(SecurityManagerTest, DISABLED_SetNULL_Intefaces) {
- security_manager_test.cc:189:TEST_F(SecurityManagerTest, DISABLED_Listeners_NULL) {
- security_manager_test.cc:198:TEST_F(SecurityManagerTest, DISABLED_Listeners_TwiceRemoveListeners) {
- security_manager_test.cc:207:TEST_F(SecurityManagerTest, DISABLED_Listeners_NoListeners) {
- security_manager_test.cc:226:TEST_F(SecurityManagerTest, DISABLED_Listeners_Notifying) {
- security_manager_test.cc:279:TEST_F(SecurityManagerTest, DISABLED_SecurityManager_NULLCryptoManager) {
- security_manager_test.cc:304:TEST_F(SecurityManagerTest, DISABLED_OnMobileMessageSent) {
- security_manager_test.cc:315:TEST_F(SecurityManagerTest, DISABLED_GetWrongServiceType) {
- security_manager_test.cc:328:TEST_F(SecurityManagerTest, DISABLED_GetEmptyQuery) {
- security_manager_test.cc:350:TEST_F(SecurityManagerTest, DISABLED_GetWrongJSONSize) {
- security_manager_test.cc:375:TEST_F(SecurityManagerTest, DISABLED_GetInvalidQueryId) {
- security_manager_test.cc:401:TEST_F(SecurityManagerTest, DISABLED_CreateSSLContext_ServiceAlreadyProtected) {
- security_manager_test.cc:416:TEST_F(SecurityManagerTest, DISABLED_CreateSSLContext_ErrorCreateSSL) {
- security_manager_test.cc:445:TEST_F(SecurityManagerTest, DISABLED_CreateSSLContext_SetSSLContextError) {
- security_manager_test.cc:479:TEST_F(SecurityManagerTest, DISABLED_CreateSSLContext_Success) {
- security_manager_test.cc:503:TEST_F(SecurityManagerTest, DISABLED_StartHandshake_ServiceStillUnprotected) {
- security_manager_test.cc:533:TEST_F(SecurityManagerTest, DISABLED_StartHandshake_SSLInternalError) {
- security_manager_test.cc:574:TEST_F(SecurityManagerTest, DISABLED_StartHandshake_SSLInitIsNotComplete) {
- security_manager_test.cc:623:TEST_F(SecurityManagerTest, DISABLED_StartHandshake_SSLInitIsComplete) {
- security_manager_test.cc:647:TEST_F(SecurityManagerTest, DISABLED_ProccessHandshakeData_WrongDataSize) {
- security_manager_test.cc:673:       DISABLED_ProccessHandshakeData_ServiceNotProtected) {
- security_manager_test.cc:708:TEST_F(SecurityManagerTest, DISABLED_ProccessHandshakeData_InvalidData) {
- security_manager_test.cc:772:TEST_F(SecurityManagerTest, DISABLED_ProccessHandshakeData_Answer) {
- security_manager_test.cc:829:TEST_F(SecurityManagerTest, DISABLED_ProccessHandshakeData_HandshakeFinished) {
- security_manager_test.cc:901:TEST_F(SecurityManagerTest, DISABLED_GetInternalError_NullData) {
- security_manager_test.cc:913:TEST_F(SecurityManagerTest, DISABLED_GetInternalError) {
- security_manager_test.cc:926:TEST_F(SecurityManagerTest, DISABLED_GetInternalError_WithErrText) {
- security_manager_test.cc:941:TEST_F(SecurityManagerTest, DISABLED_GetInternalError_WithErrJSONText) {

This PR contains changes and review notes fixes from [previous PR](https://github.com/OHerasym/sdl_core/pull/3).

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @anosach-luxoft , @dev-gh, @AByzhynar

File: src/components/protocol_handler/test/multiframe_builder_test.cc [hunk 1]
```diff
@@ -37,11 +37,16 @@
...
 #include "utils/make_shared.h"
 #include "protocol_handler/multiframe_builder.h"
+#include "utils/threads/thread.h"
 
 namespace test {
 namespace components {
 namespace protocol_handler_test {
 
+namespace {
+const int32_t MICROSECONDS_IN_MILLISECONDS = 1000;
```
Comment (anosach-luxoft) [inline src/components/protocol_handler/test/multiframe_builder_test.cc]: @OHerasym include `date_time` with this const

File: src/components/protocol_handler/test/multiframe_builder_test.cc [hunk 2]
```diff
@@ -506,7 +512,7 @@ TEST_F(MultiFrameBuilderTest, FrameExpired_OneMSec) {
       << "First frame: " << first_frame;
 
   // Wait frame expire
-  usleep(1000);
+  threads::sleep(1000);
```
Comment (anosach-luxoft) [inline src/components/protocol_handler/test/multiframe_builder_test.cc]: @OHerasym `usleep` receives microseconds, but `thread::sleep` milliseconds

File: src/components/protocol_handler/test/protocol_handler_tm_test.cc [hunk 3]
```diff
@@ -299,13 +299,13 @@ class OnHandshakeDoneFunctor {
 /*
  * ProtocolHandler shall skip empty message
  */
-TEST_F(ProtocolHandlerImplTest, RecieveEmptyRawMessage) {
```
Comment (anosach-luxoft) [inline src/components/protocol_handler/test/protocol_handler_tm_test.cc:302]: @OHerasym just disable whole file in `CMakeLists`

File: src/components/security_manager/test/CMakeLists.txt [hunk 4]
```diff
@@ -68,7 +68,19 @@ foreach( file_i ${CERT_LIST})
...
+endif()
+
+if(NOT QT_PORT AND ${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  link_directories(${CMAKE_SOURCE_DIR}/build/openssl_win_x64/lib/)
+endif()
+
 create_test (security_manager_test "${SOURCES}" "${LIBRARIES}")
 add_dependencies(security_manager_test generate_certificates)
 
+if(MSVC)
```
Comment (anosach-luxoft) [inline src/components/security_manager/test/CMakeLists.txt]: @OHerasym `WIN_NATIVE`

File: src/components/security_manager/test/crypto_manager_impl_test.cc [hunk 5]
```diff
@@ -29,7 +29,9 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-
+#if defined(OS_WINDOWS)
+#include <WinSock2.h>
```
Comment (anosach-luxoft) [inline src/components/security_manager/test/crypto_manager_impl_test.cc]: @OHerasym `winhdr.h`

File: src/components/security_manager/test/security_manager_test.cc [hunk 6]
```diff
@@ -170,7 +170,9 @@ class SecurityManagerTest : public ::testing::Test {
  * SecurityManager shall not set NULL interfaces
  * and shall not call any methodes
  */
-TEST_F(SecurityManagerTest, SetNULL_Intefaces) {
```
Comment (anosach-luxoft) [inline src/components/security_manager/test/security_manager_test.cc:173]: @OHerasym the same. Just comment whole file

File: src/components/security_manager/test/ssl_context_test.cc [hunk 7]
```diff
@@ -29,7 +29,9 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-
+#if defined(OS_WINDOWS)
+#include <WinSock2.h>
```
Comment (anosach-luxoft) [inline src/components/security_manager/test/ssl_context_test.cc]: @OHerasym `winhdr.h`

File: src/components/transport_manager/include/transport_manager/tcp/tcp_client_listener.h [hunk 8]
```diff
@@ -100,6 +100,16 @@ class TcpClientListener : public ClientConnectionListener {
    */
   virtual TransportAdapter::Error StopListening();
 
+#ifdef BUILD_TESTS
+  uint16_t port() const {
+    return port_;
+  }
+
+  threads::Thread* thread() const {
```
Comment (anosach-luxoft) [inline src/components/transport_manager/include/transport_manager/tcp/tcp_client_listener.h:108]: @OHerasym what for do you need this?

File: src/components/transport_manager/test/CMakeLists.txt [hunk 9]
```diff
@@ -76,4 +81,10 @@ set(SOURCES
 
 create_test("transport_manager_test" "${SOURCES}" "${LIBRARIES}")
 file(COPY smartDeviceLink_test.ini DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+
+if(MSVC)
```
Comment (anosach-luxoft) [inline src/components/transport_manager/test/CMakeLists.txt]: @OHerasym `WIN_NATIVE`

File: src/components/transport_manager/test/tcp_transport_adapter_test.cc [hunk 10]
```diff
@@ -259,34 +254,38 @@ TEST_F(TcpAdapterTest, StoreData_ConnectionNotExist_DataNotStored) {
...
+      MakeShared<MockTCPDevice>(host_address, uniq_id);
   transport_adapter.AddDevice(mockdev);
 
   std::vector<std::string> devList = transport_adapter.GetDeviceList();
   ASSERT_EQ(1u, devList.size());
   EXPECT_EQ(uniq_id, devList[0]);
 
   const int app_handle = 1;
-  std::vector<int> intList = {app_handle};
+  std::vector<int> intList;
```
Comment (AByzhynar) [inline src/components/transport_manager/test/tcp_transport_adapter_test.cc]: @OHerasym May be use vector ctor with parameters if you just add 1 element? I mean `std::vector<int> intList(1, app_handle)` 
What do you think?

**Comment by AByzhynar** [comment]
@OHerasym **Add please link to related issue**

File: src/components/transport_manager/include/transport_manager/tcp/tcp_client_listener.h (hunk 8 above)
Comment (OHerasym) [inline src/components/transport_manager/include/transport_manager/tcp/tcp_client_listener.h:108]: @anosach-luxoft , is used in TEST_F(TcpClientListenerTest, Ctor_test) test

File: src/components/protocol_handler/test/multiframe_builder_test.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/protocol_handler/test/multiframe_builder_test.cc]: @anosach-luxoft , removed unused var

File: src/components/protocol_handler/test/multiframe_builder_test.cc (hunk 2 above)
Comment (OHerasym) [inline src/components/protocol_handler/test/multiframe_builder_test.cc]: @anosach-luxoft , Done

File: src/components/protocol_handler/test/protocol_handler_tm_test.cc (hunk 3 above)
Comment (OHerasym) [inline src/components/protocol_handler/test/protocol_handler_tm_test.cc:302]: @anosach-luxoft , Done

File: src/components/security_manager/test/CMakeLists.txt (hunk 4 above)
Comment (OHerasym) [inline src/components/security_manager/test/CMakeLists.txt]: @anosach-luxoft , changed to ${CMAKE_SYSTEM_NAME} MATCHES "Windows"

File: src/components/security_manager/test/crypto_manager_impl_test.cc (hunk 5 above)
Comment (OHerasym) [inline src/components/security_manager/test/crypto_manager_impl_test.cc]: @anosach-luxoft , Done

File: src/components/security_manager/test/security_manager_test.cc (hunk 6 above)
Comment (OHerasym) [inline src/components/security_manager/test/security_manager_test.cc:173]: @anosach-luxoft , Done

File: src/components/security_manager/test/ssl_context_test.cc (hunk 7 above)
Comment (OHerasym) [inline src/components/security_manager/test/ssl_context_test.cc]: @anosach-luxoft , Done

File: src/components/transport_manager/test/CMakeLists.txt (hunk 9 above)
Comment (OHerasym) [inline src/components/transport_manager/test/CMakeLists.txt]: @anosach-luxoft , changed to ${CMAKE_SYSTEM_NAME} MATCHES "Windows"

File: src/components/transport_manager/test/tcp_transport_adapter_test.cc (hunk 10 above)
Comment (OHerasym) [inline src/components/transport_manager/test/tcp_transport_adapter_test.cc]: @AByzhynar , Done

**Comment by OHerasym** [comment]
@AByzhynar , added link to related issue

**Comment by OHerasym** [comment]
@AByzhynar , @anosach-luxoft , Please review changes

**Pull request merged**